lib/StackAnalysis/StackAnalysis.cpp

```diff
@@ -441,6 +1975,244 @@ bool StackAnalysis::runOnModule(Module &M) {
+  }
+
+  for (CFEP &Function : Functions) {
+    llvm::SmallSet<BasicBlock *, 8> Visited;
+    auto *StartNode = BasicBlockNodeMap[Function.Entry];
+    Worklist.emplace_back(Function.Entry);
+
+    do {
+      auto *Current = Worklist.pop_back_val();
+      Visited.insert(Current);
+      for (auto *Successor : successors(Current)) {
+        if (isFunctionCall(Current)) {
+          auto *Callee = getFunctionCallCallee(Current);
+          // If not an indirect call, add the node to the CG
+          if (Callee) {
+            auto *Node = BasicBlockNodeMap[Callee];
+            StartNode->addSuccessor(Node);
+          }
+          auto *Next = getFallthrough(Current);
+          if (!Visited.count(Next))
+            Worklist.push_back(Next);
+          continue;
+        }
+
+        if (!GCBI::isPartOfRootDispatcher(Successor)
+            && !Visited.count(Successor))
+          Worklist.push_back(Successor);
+      }
+    } while (!Worklist.empty());
+  }
+
+  for (const auto &[_, Node] : BasicBlockNodeMap) {
+    if (!Node->hasPredecessors())
+      EntryBlocks.emplace_back(Node);
+
+    // Recursive?
+    if (*Node->predecessors().begin() == Node)
+      EntryBlocks.emplace_back(Node);
+  }
+
+  BasicBlockNode FakeNode{ nullptr };
+  auto *RootNode = CG.addNode(FakeNode);
+  CG.setEntryNode(RootNode);
+  for (const auto &Node : EntryBlocks)
+    RootNode->addSuccessor(Node);
+
+  if (StackAnalysisLog.isEnabled()) {
+    std::error_code EC;
+    raw_fd_ostream Out(M.getModuleIdentifier() + "_CG.dot",
+                       EC,
+                       llvm::sys::fs::OF_None);
+    revng_assert(!EC);
+    llvm::WriteGraph(Out, &CG);
+    revng_log(StackAnalysisLog, "Call graph written on disk");
+    Out.close();
+  }
+
+  for (auto *Node : llvm::post_order(&CG)) {
+    if (Node->hasPredecessors())
+      CFEPQueue.insert(Node);
+  }
+
+  // Collect all the ABI registers
+  std::vector<llvm::GlobalVariable *> ABIRegisters;
+  for (auto *CSV : GCBI.abiRegisters())
+    if (CSV && !(GCBI.isSPReg(CSV)))
+      ABIRegisters.emplace_back(CSV);
+
+  std::optional<int64_t> FSO;
+  switch (GCBI.arch()) {
+  case llvm::Triple::x86: {
+    FSO = 4;
+    break;
+  }
+  case llvm::Triple::x86_64: {
+    FSO = 8;
+    break;
+  }
+  default: {
+    FSO = 0;
+    break;
+  }
+  }
+
+  // Default-constructed cache summary for indirect calls
+  FunctionSummary DefaultSummary(model::FunctionType::Values::Regular,
+                                 { ABIRegisters.begin(), ABIRegisters.end() },
+                                 ABIAnalyses::ABIAnalysesResults(),
+                                 {},
+                                 FSO,
+                                 nullptr);
+  FunctionAnalysisResults Properties(std::move(DefaultSummary));
+
+  // Open streams for dumping results
+  std::unique_ptr<raw_fd_ostream> OutputIBI;
+  if (IndirectBranchInfoSummaryPath.getNumOccurrences() == 1) {
+    std::remove(IndirectBranchInfoSummaryPath.c_str());
```

Comment:
You're ignoring the return value.

Reply:
I understand, but look, it's really the return value of `std::remove`. Like who cares about `std::remove`'s rv? More specifically, 1) who cares if it exists? The user might have created it in the past, if so we remove it, otherwise we just don't; 2) does the user need to know it was not removed b/c it didn't exist? That said, I'm changing to check the return value. And I really don't wanna sound annoying, but it just doesn't matter.

Reply:
`std::remove` does not fail only in case the file does not exist.
You might not be allowed to delete the fail, but you might be allowed to write it.